src/cmd/ksh93/bltins/print.c

```diff
@@ -93,33 +93,24 @@ static char *nullarg[] = {0, 0};
 // Builtin `echo`.
 //
 int B_echo(int argc, char *argv[], Shbltin_t *context) {
-    static char bsd_univ;
     struct print prdata;
     prdata.options = sh_optecho + 5;
     prdata.raw = prdata.echon = 0;
     prdata.sh = context->shp;
     UNUSED(argc);
 
-    // This mess is because /bin/echo on BSD is different.
-    if (!prdata.sh->universe) {
-        char *universe;
-        universe = astconf("UNIVERSE", 0, 0);
-        if (universe) bsd_univ = (strcmp(universe, "ucb") == 0);
-        prdata.sh->universe = 1;
-    }
-    if (!bsd_univ) return b_print(0, argv, (Shbltin_t *)&prdata);
+    // This mimics BSD style echo command
+    // See https://github.com/att/ast/issues/370
     prdata.options = sh_optecho;
     prdata.raw = 1;
```

Comment:
I was worried that selecting the BSD variant would not support `-e` since the macOS external `echo` doesn't support it. Since the ksh BSD behavior does support it this change LGTM. However, I would hoist this to line 98 -- it's kind of silly to init this var twice:

```
prdata.echon = 0;
prdata.raw = 1;
```

rather than the existing

```
prdata.raw = prdata.echon = 0;
```